programs/uxd/tests/integration_tests/api/program_uxd/instructions/process_collect_profits_of_mercurial_vault_depository.rs

```diff
@@ -0,0 +1,233 @@
+        collateral_mint,
+        &mercurial_vault,
+    )
+    .0;
+    let mercurial_vault_depository_lp_token_vault =
+        program_uxd::accounts::find_mercurial_vault_depository_lp_token_vault_pda(
+            collateral_mint,
+            &mercurial_vault,
+        )
+        .0;
+    let mercurial_vault_collateral_token_safe =
+        program_mercurial::accounts::find_token_vault_pda(&mercurial_vault).0;
+
+    // Read state before
+    let controller_before =
+        program_test_context::read_account_anchor::<Controller>(program_test_context, &controller)
+            .await?;
+    let mercurial_vault_depository_before = program_test_context::read_account_anchor::<
+        MercurialVaultDepository,
+    >(program_test_context, &mercurial_vault_depository)
+    .await?;
+    let mercurial_vault_before =
+        program_test_context::read_account_anchor::<mercurial_vault::Vault>(
+            program_test_context,
+            &mercurial_vault,
+        )
+        .await?;
+    let mercurial_vault_lp_mint_before = program_test_context::read_account_packed::<Mint>(
+        program_test_context,
+        mercurial_vault_lp_mint,
+    )
+    .await?;
+
+    let mercurial_vault_depository_lp_token_vault_amount_before =
+        program_test_context::read_account_packed::<Account>(
+            program_test_context,
+            &mercurial_vault_depository_lp_token_vault,
+        )
+        .await?
+        .amount;
+
+    let profits_beneficiary_collateral_amount_before =
+        program_test_context::read_account_packed::<Account>(
+            program_test_context,
+            profits_beneficiary_collateral,
+        )
+        .await?
+        .amount;
+
+    let unix_timestamp_before = u64::try_from(
+        program_test_context
+            .banks_client
+            .get_sysvar::<Clock>()
+            .await
+            .unwrap()
+            .unix_timestamp,
+    )
+    .unwrap();
+
+    // Execute IX
+    let accounts = uxd::accounts::CollectProfitsOfMercurialVaultDepository {
+        payer: payer.pubkey(),
+        controller,
+        collateral_mint: *collateral_mint,
+        depository: mercurial_vault_depository,
+        depository_lp_token_vault: mercurial_vault_depository_lp_token_vault,
+        mercurial_vault,
+        mercurial_vault_lp_mint: *mercurial_vault_lp_mint,
+        mercurial_vault_collateral_token_safe,
+        profits_beneficiary_collateral: *profits_beneficiary_collateral,
+        system_program: anchor_lang::system_program::ID,
+        token_program: anchor_spl::token::ID,
+        mercurial_vault_program: mercurial_vault::ID,
+    };
+    let payload = uxd::instruction::CollectProfitsOfMercurialVaultDepository {};
+    let instruction = Instruction {
+        program_id: uxd::id(),
+        accounts: accounts.to_account_metas(None),
+        data: payload.data(),
+    };
+    program_test_context::process_instruction(program_test_context, instruction, payer).await?;
```

Comment:
nit: do not hesitate to use `scopes` in order to split the function gracefully and improve readability :)

Reply:
Yhea Greg's scopes are pretty neat I should use that more too

Reply:
Ah yes I often forget about those, good point.

So i went around in the PR and tried to look into this.

There are a few patterns I was looking for:

### Pattern A

```rs
let my_final_result = {
    let my_inner_variable;
    my_computation_that_returns_value(my_inner_variable)
}
```

### Pattern B

```rs
{
   my_computation_that_doesnt_return_value();
   my_computation_that_doesnt_return_value();
}
```

### Pattern C

```rs
{
   let my_inner_variable_not_needed_outside;
   my_computation_without_return_value(my_inner_variable_not_needed_outside);
}
```

### Pattern D

```rs
let my_value_used_everywhere;
// later
{
    my_computation1(my_value_used_everywhere)
}
// later
{
   my_computation1(my_value_used_everywhere)
}
```

### Looking at the PR:

I wasnt able to find any pattern A which in my opinion is very useful.

Only pattern B and C (or D) generally accross the codebase, in which case the scopes are less useful imo.

I can add more scopes in future incremental PRs no problem, I kinda like the concept, but I havent found places where they would be super useful yet.

I am curious as to where you would see them fit in this case? I like code separation so im looking forward to adding more.

I just think pattern B,C,D are not good enough to warrant them honestly because of the "no unecessary indent" rule.

A quick explaination of my point of view on identation here: https://www.youtube.com/watch?v=CFRhGnuXG-4

Anyhow, would love to hear some example from you, I like scope, i just havent found many good places to use them.

(And I dont like pattern B for example)

Reply:
Ima merge it for now, but feel free to reply to this thread, I will adjust in future incremental PRs